streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1374,7 +1376,7 @@ public void signalResume() {
      */
     int maybeCommit() {
         final int committed;
-        if (now - lastCommitMs > commitTimeMs) {
+        if (taskManager.needsCommit(true) || now - lastCommitMs > commitTimeMs) {
```

Comment:
I've added a test for this in 0577905, but it feels like a bit of a tautology: it essentially just tests that `commit()` is called when `taskManager.needsCommit(anyBoolean())` returns `true`.

The real meat of this will be in the `TaskManagerTests`, which I'm working on separately.

Reply:
We need both tests. The one in `StreamThreadTest` verifies that the commit is triggered or not depending on the value returned from `needsCommit()` and the one in `TaskManagerTest` verifies that `needCommit()` returns the expected boolean value.

Apropos `TaskManagerTest`, I missing a unit test for `needsCommit()` there. Good that you are already working on it.